### PR TITLE
DOAB Fetcher now fetches ISBN and imprint fields where possible

### LINKED ISSUE #8708: DOAB fetcher does not fetch field ISBN
### JabRef version

Latest development branch build (please note build date below)

### Operating system

Windows

### Details on version and operating system

_No response_

### Checked with the latest development build

- [X] I made a backup of my libraries before testing the latest development version.
- [X] I have tested the latest development version and the problem persists

### Steps to reproduce the behaviour

See this entry: https://directory.doabooks.org/handle/20.500.12854/35818

Search for "Accounts and Accountability in Late Medieval Europe" in JabRef DOAB websearch.

It finds:

```
@Book{EpurescuPascovici2020aaa,
  date      = {2020},
  title     = {Accounts and Accountability in Late Medieval Europe},
  doi       = {10.1484/M.USML-EB.5.119569},
  editor    = {Ionuț Epurescu-Pascovici},
  language  = {English},
  pages     = {318},
  publisher = {Brepols},
  subtitle  = {Records, Procedures, and Socio-Political Impact},
  url       = {http://library.oapen.org/handle/20.500.12657/39752},
  abstract  = {This volume stems from a conference organised in November 2018 at the
University of Bucharest in the frame of the European Research Council
project CASTELLANY ACCOUNTS, “Record-keeping, fiscal reform, and
the rise of institutional accountability in late-medieval Savoy: A source-oriented
approach”. Both the conference and the volume have received funding
from the European Research Council (ERC) under the European Union’s Horizon
2020 research and innovation programme (grant agreement No. 638436).
The volume traces the momentous transformation of institutions and administration under the impact of accounting records and procedures, c. 1250-1500. The focus on the materiality and organising logic of a range of accounts is complemented by close attention to the socio-political contexts in which they functioned and the agency of central and local officials.},
  keywords  = {middle ages, Europe, auditing, accounts, HB},
  uri       = {https://directory.doabooks.org/handle/20.500.12854/35818},
}
```

The ISBN field is missing.

Additional info:

code can be found here: [src/main/java/org/jabref/logic/importer/fetcher/DOABFetcher.java](https://github.com/JabRef/jabref/pull/8598/files#diff-5211ecb27db75d41827a8e4461c4fa0acd0cc152748d79da8a3dc1318b10d7fe)

### Appendix

_No response_

### PR DESCRIPTION
**Pull request by shafinkamal**
Fixes #8708.

Previously, the DOAB Fetcher could not obtain the ISBN of a DOAB entry. This PR modifies the URL for DOAB query to include `BITSTREAM` data (where ISBN information can be found). 

Below is an example of the previous and updated URL's for DOAB queries (note the ",bitstreams" suffix)

`PREVIOUS`: https://directory.doabooks.org/rest/search?query=handle:%2220.500.12854/26303%22&expand=metadata

`UPDATED`: https://directory.doabooks.org/rest/search?query=handle:%2220.500.12854/26303%22&expand=metadata,bitstreams

The fetcher can then obtain the ISBN and add the field to the entry accordingly. I am creating this Draft PR to discuss with maintainers that it seems like **_many_** DOAB entries do not have ISBN metadata or `BITSTREAM` objects and so the ISBN may perhaps be unobtainable in these instances. So I am wondering if there is anything more can be done or if (under the assumption that this PR is accepted firstly) #8708 should be closed?

- [x] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [x] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [x] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [x] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

**Comment by ThiloteE** [comment]
When DOAB entries lack ISBN, then obviously JabRef can't fetch them. That's fine then. No more work needed to be done.

As long as your changes do not break anything and JabRef now can fetch the `ISBN` and the `imprint` fields, issue 8708 can indeed be closed.

**Comment by shafinkamal** [comment]
Thanks for clarifying, @ThiloteE.

To follow up, which `StandardField` should the `imprint` field be associated with? The only related `StandardField` I see is `PUBLISHER` which is already associated with is `publisher.name`.

Thanks again :)

**Comment by ThiloteE** [comment]
You are correct. Nice catch. `imprint` is indeed not a bibtex or biblatex standard field. Using https://en.wikipedia.org/wiki/Imprint_(trade_name) as basis and comparing with bibtex and biblatex standards, the bibtex / biblatex `Publisher` field indeed seems to be the closest to it.

So, I would introduce a "If the publisher field is empty, then map imprint to publisher" condition

I found an entry for testing:

https://directory.doabooks.org/handle/20.500.12854/70984 has both imprint and publisher.

I have not found an entry without publisher yet.

**Comment by shafinkamal** [comment]
Hello, I am seeking advice/recommendations on how to resolve the failing checks. The 'Fetcher Tests' check seems to fail at `DBLPFetcherTest` which from my understanding, is not related to the commits I have made. 

Also unsure on how to resolve the check: 'Deployment / Create installer and portable version for macOS'.

Any help is greatly appreciated.

Thanks!

**Comment by ThiloteE** [comment]
You can ignore those two. The fetcher error is a http error code 500, which points to serverside error and to compile the macOS version of JabRef there has to be done something related to certificates and/or injecting secrets, which usually is only done by maintainers for a proper release of JabRef.

**Comment by shafinkamal** [comment]
Thanks @ThiloteE. Marking the PR for review.

File: src/main/java/org/jabref/logic/importer/fetcher/DOABFetcher.java [hunk 1]
```diff
@@ -123,15 +146,24 @@ private BibEntry jsonToBibEntry(JSONArray metadataArray) {
...
             }
         }
+
         entry.setField(StandardField.AUTHOR, AuthorList.of(authorsList).getAsFirstLastNamesWithAnd());
         entry.setField(StandardField.EDITOR, AuthorList.of(editorsList).getAsFirstLastNamesWithAnd());
         entry.setField(StandardField.KEYWORDS, String.valueOf(keywordJoiner));
+
+        // Special condition to check if publisher field is empty. If so, retrieve imprint (if available)
+        if (entry.getField(StandardField.PUBLISHER).isEmpty()) {
+            if (!publisherImprint.equals("")) {
```
[the remark this answers is not in the document]
Comment (Siedlerchr) [reply src/main/java/org/jabref/logic/importer/fetcher/DOABFetcher.java:163]: You can use that from model.StringUtil

**Review by Siedlerchr — APPROVED** [review]
Thanks for your contribution! Overall looks good to me! Just a tiny improvement

**Comment by shafinkamal** [comment]
Thanks @Siedlerchr. Committed + pushed the changes.

**Review by calixtus — APPROVED** [review]

**Comment by calixtus** [comment]
Thanks @shafinkamal for you contribution. Just merged it! Looking forward to see more 👍

**Pull request merged**